3.0/vector_tile.proto

```diff
@@ -44,6 +44,63 @@ message Tile {
                 // A detailed description on geometry encoding is located in 
                 // section 4.3 of the specification.
                 repeated uint32 geometry = 4 [ packed = true ];
+                
+                // Additional attributes (or all the attributes) of this feature may be
+                // encoded as repeated pairs of 64-bit integers, to take
+                // advantage of inline encoding of small values,
+                // improved compression from use of repeated values,
+                // and support for list and map values.
+                //
+                // This message may only be used if the layer version is >= 3.
+                //
+                // The inline_attributes field is much like the attributes field in that it is a pair of
+                // integers that reference key and value pairs. However, the value reference
+                // is a "complex_value" that combines a type and an index.
+                // 
+                // The "key_index" is much like the key index in the use for attributes, but instead
+                // of pointing into layer.keys, it points into layer.attribute_pool.keys.
+                //
+                // An complex value has two parts: the lowest 4 bits are the type bits,
+                // and the remaining bits are the parameter bits. What is stored in the parameter
+                // bits is dependent on the contents of the type bits. For example for inline values,
+                // the parameter field is not an index but simply a value. For other types it is
+                // an index position into a value storage of the layer.
+                // 
+                // uint64_t type = complex_value & 0x0F; // Bottom 4 bits
+                // uint64_t parameter = complex_value >> 4;
+                // 
+                //     Type     | Id  | Parameter
+                // ---------------------------------
+                // string       |  0  | index into layer.attribute_pool.string_values
+                // float        |  1  | index into layer.attribute_pool.float_values
+                // double       |  2  | index into layer.attribute_pool.double_values
+                // int          |  3  | index into layer.attribute_pool.signed_integer_values
+                // uint         |  4  | index into layer.attribute_pool.unsigned_integer_values
+                // inline uint  |  5  | value of unsigned integer (values between 0 to 2^60-1)
+                // inline sint  |  6  | value of zigzag-encoded integer (values between -2^59 to 2^59-1)
```

Comment:
We probably should change this to be 2^56 for uint and 2^55  for signed due to the way varints are encoded.